.github/actions/breaking-pr-check/index.js

```diff
@@ -0,0 +1,60 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+
+function raiseError(message) {
+  throw new Error(message);
+}
+
+async function getPullRequest() {
+  const client = github.getOctokit(process.env.GITHUB_TOKEN);
+
+  const pr = github.context.payload.pull_request;
+  if (!pr) {
+    throw new Error(
+      "This action can only be invoked in `pull_request_target` or `pull_request` events. Otherwise the pull request can't be inferred.",
+    );
+  }
+
+  const owner = pr.base.user.login;
+  const repo = pr.base.repo.name;
+
+  const { data } = await client.rest.pulls.get({
+    owner,
+    repo,
+    pull_number: pr.number,
+  });
+
+  return data;
+}
+
+function checkTitle(title) {
+  if (/!/.test(title)) {
+    raiseError(`Do not use exclamation mark ('!') in your PR Title.`);
+  }
+}
```

Comment:
Hmmm, isn't this too restrictive? I can imagine the several cases when `!` might be used not in the "breaking changes" position (before the semicolon)

Here are several real world PRs that have an exclamation mark in their titles:
- https://github.com/typescript-eslint/typescript-eslint/pull/5594
- https://github.com/typescript-eslint/typescript-eslint/pull/6968
- https://github.com/typescript-eslint/typescript-eslint/pull/1460

I guess we should restrict only the use of `feat(...)!: ...` and `feat!: ...`. What do you think?